### ISSUE TITLE
Proposal: Add IsReadOnly / IsValueLocked property to NumberBox

### ISSUE THREAD

**Issue by MuziburRahman** [microsoft/microsoft-ui-xaml#2204]
The `NumberBox` control is very useful. 

In many cases I want to make the property that's bound to it's `ValueProperty` unchangeable, keeping it readable, which is what the `IsReadOnly` property does with the `TextBox` control.

If it had an event like the `TextChanging` event of `TextBox`, there could be a work around, but it doesn't have that.

**Comment by StephenLPeters** [comment]
If you are desperate you could use visualtree helper to walk down to the inner textbox and attach to its TextChanging event. We'll consider this going foreword.

**Comment by myokeeh** [comment]
I'd like to see `IsReadOnly`, too

**Comment by StephenLPeters** [comment]
agreed, should be easy, of note though, https://github.com/microsoft/microsoft-ui-xaml/issues/2672 IsReadOnly on textbox doesn't offer any affordance of read only.

**Comment by robloo** [comment]
@StephenLPeters Well, the best we can do is match the behavior of the TextBox and I think that's fine for this. If/when TextBox is improved the NumberBox would automatically get the same improvements as it will just forward to the InputBox.

The only added complication I thought of for the NumberBox is we will have to adjust the visual states. The up/down buttons will need to be disabled as well.

**Comment by MuziburRahman** [comment]
How about some kind of lock icon that will be visible when it is locked? 
Like this:

![image](https://user-images.githubusercontent.com/21083609/91929781-f7b77600-ed00-11ea-9747-c0e7bc181018.png)

This way user can easily distinguish it with editable NumberBoxes. Besides, if the spin buttons are inlined, then in 'IsReadOnly' mode, those spins buttons won't be visible anymore. That's another indicator.

**Comment by robloo** [comment]
@MuziburRahman It looks nice but no other control uses this convention. I think the lock button should be a proposal for the TextBox to start. Such visual cues should really be uniform in all of WinUI. However, for some controls this convention wouldn't always work so well -- when existing search or calendar icons are present for example.
